### PR TITLE
Make Gradle build fully cacheable

### PR DESCRIPTION
**Pull request by serandel**
As a new hire in Gradle, part of my onboarding has been updating the build of an OSS project to be fully cacheable and we've chosen Groovy for this purpose. Also, the Gradle wrapper and the Gradle Enterprise plug-in have been upgraded to the latest released version. 

Please, have a look at my proposed changes, and merge them if you agree with them. :)

### CODE REVIEW

**Comment by danielsun1106** [comment]
LGTM.
According to the Apache rules, binary files are not allowed to include in codebase, e.g. gradle/wrapper/gradle-wrapper.jar

**Comment by paulk-asert** [comment]
> According to the Apache rules, binary files are not allowed to include in codebase, e.g. gradle/wrapper/gradle-wrapper.jar

Actually, the jar can be in the codebase but must not be in a release (in particular the source release). So, that part of the PR looks okay to me. We already have bootstrap steps described in the README for getting that jar when using the source release.

Also, we track dependency and tool version changes via a new issue in Jira. Would you be able to create that @serandel?

**Comment by danielsun1106** [comment]
Merged. Thanks!

**Comment by serandel** [comment]
> > According to the Apache rules, binary files are not allowed to include in codebase, e.g. gradle/wrapper/gradle-wrapper.jar
> 
> Actually, the jar can be in the codebase but must not be in a release (in particular the source release). So, that part of the PR looks okay to me. We already have bootstrap steps described in the README for getting that jar when using the source release.
> 
> Also, we track dependency and tool version changes via a new issue in Jira. Would you be able to create that @serandel?

It would be a pleasure, of course. 

https://issues.apache.org/jira/browse/GROOVY-9942

Can you please have a look at the Fix Version and edit it if it's not the right one, please? Thanks!

File: gradle/verification-metadata.xml [hunk 1]
```diff
@@ -1,42 +1,4 @@
...
-   troubleshooting guide at https://docs.gradle.org/current/userguide/dependency_verification.html
-
-   This build uses PGP verification and SHA512 checksum verification for unsigned artifacts:
-
-   ./gradlew -write-verification-metadata sha512,pgp -export-keys <tasks>
-
-   Then check the diff, remove the unnecessary entries (for old dependencies) and make sure
-   the new ones are correct.
-
-   Do not remove this header.
```
Comment (melix) [inline gradle/verification-metadata.xml:37]: :x: This says "Do not remove this header" and the PR removes the header. It shows that the file hasn't been reviewed, when it should ;)

Comment (danielsun1106) [reply gradle/verification-metadata.xml:37]: Good catch ;-)

**Pull request closed without merging**